### PR TITLE
Implement server side cluster resolving

### PR DESCRIPTION
**Pull request by jysohn23**
```
.......................
----------------------------------------------------------------------
Ran 23 tests in 0.428s

OK
Name                                         Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------
/pytorch/xla/torch_xla_py/xla_dist.py          173     12    93%   12-13, 52, 82, 172, 175, 186-189, 239, 251, 283, 326
/pytorch/xla/torch_xla_py/xla_dist_test.py     257      4    98%   249, 267-268, 734
--------------------------------------------------------------------------
TOTAL                                          430     16    96%
```

Also applied Google `pyformat` again.

### CODE REVIEW

File: torch_xla_py/xla_dist.py [hunk 1]
```diff
@@ -21,16 +21,35 @@
 class Worker(object):
 
   def __init__(self, internal_ip, machine_type, zone):
-    self._internal_ip = internal_ip
-    self._machine_type = machine_type
-    self._zone = zone
+    self._internal_ip = str(internal_ip)
+    self._machine_type = str(machine_type)
+    self._zone = str(zone)
```
Comment (taylanbil) [inline torch_xla_py/xla_dist.py]: why are these str?

Comment (jysohn23) [reply torch_xla_py/xla_dist.py]: I added them to be string to be consistent to consider different types. More significant for stuff like `port` where I realized that `"80"` != `80`.

Comment (dlibenzi) [reply torch_xla_py/xla_dist.py]: They should not be needed. They should all be string in input, and if not, it is call site error.

File: torch_xla_py/xla_dist.py [hunk 2]
```diff
@@ -21,16 +21,35 @@
...
 
 
 class ClientWorker(Worker):
 
   def __init__(self, internal_ip, machine_type, zone, hostname=None):
     super(ClientWorker, self).__init__(internal_ip, machine_type, zone)
-    self._hostname = hostname
+    self._hostname = str(hostname)
```
Comment (dlibenzi) [inline torch_xla_py/xla_dist.py]: Same here.

File: torch_xla_py/xla_dist.py [hunk 3]
```diff
@@ -53,8 +72,34 @@ def __hash__(self):
 
 
 class ServiceWorker(Worker):
-  # Same as base Worker ATM.
-  pass
+
+  def __init__(self, internal_ip, port, machine_type, zone, sw_version):
+    super(ServiceWorker, self).__init__(internal_ip, machine_type, zone)
+    self._port = str(port)
```
Comment (dlibenzi) [inline torch_xla_py/xla_dist.py]: Here, you can eventually do an `int()` and enforce that a port must be integer.

File: torch_xla_py/xla_dist.py [hunk 4]
```diff
@@ -53,8 +72,34 @@ def __hash__(self):
...
+    self._sw_version = str(sw_version)
```
Comment (dlibenzi) [inline torch_xla_py/xla_dist.py]: No `str()` here.

File: torch_xla_py/xla_dist.py (hunk 1 above)
Comment (jysohn23) [reply torch_xla_py/xla_dist.py]: Done.

File: torch_xla_py/xla_dist.py (hunk 2 above)
Comment (jysohn23) [reply torch_xla_py/xla_dist.py]: Done.

File: torch_xla_py/xla_dist.py (hunk 3 above)
Comment (jysohn23) [reply torch_xla_py/xla_dist.py]: Done.

File: torch_xla_py/xla_dist.py (hunk 4 above)
Comment (jysohn23) [reply torch_xla_py/xla_dist.py]: Done.

**Pull request merged**